urlencoder-app/build.gradle.kts

```diff
@@ -45,14 +45,13 @@ tasks {
 
     val fatJar by registering(Jar::class) {
         group = LifecycleBasePlugin.BUILD_GROUP
-        dependsOn.addAll(listOf("compileJava", "compileKotlinJvm", "processResources"))
         archiveClassifier.set("all")
         duplicatesStrategy = DuplicatesStrategy.EXCLUDE
-        manifest { attributes(mapOf("Main-Class" to application.mainClass)) }
-        from(sourceSets.main.get().output)
+        manifest { attributes(mapOf("Main-Class" to urlEncoderMainClass)) }
+        from(sourceSets.main.map { it.output })
         dependsOn(configurations.jvmRuntimeClasspath)
         from(configurations.jvmRuntimeClasspath.map { classpath ->
-            classpath.incoming.artifacts.artifactFiles.files.filter { it.name.endsWith("jar") }.map { zipTree(it) }
+            classpath.filter { it.name.endsWith(".jar") }.map { zipTree(it) }
         })
```

Comment:
These changes are some unrelated tidying, to make the config a bit neater, and make it more Gradle idiomatic.